packages/aws-rfdk/lib/deadline/test/version.test.ts

```diff
@@ -13,11 +13,153 @@ import {
+      ], [
+        'less than',
+        {
+          firstVersion: '1.1.1.1',
+          secondVersion: '1.1.1.2',
+          expectedValue: false,
+        },
+      ], [
+        'greater than',
+        {
+          firstVersion: '2.0.0.1',
+          secondVersion: '2.0.0.0',
+          expectedValue: true,
+        },
+      ],
+    ])('%s', (_name, testcase) => {
+      const { firstVersion, secondVersion, expectedValue } = testcase;
+
+      // WHEN
+      const lhs = Version.parse(firstVersion);
+      const result = lhs.isGreaterThan(Version.parse(secondVersion));
+
+      expect(result).toEqual(expectedValue);
+    });
+  });
+
+  describe('.isGreaterThan constructor', () => {
+
+    // WHEN
+    const lhs = new Version([10, 1, 9, 2]);
+    const result = lhs.isGreaterThan(Version.parse('10.0.9.2'));
+
+    expect(result).toEqual(true);
+
+  });
+
+  describe('.isLessThan', () => {
+    test.each<[string, { firstVersion: string, secondVersion: string, expectedValue: boolean }]>([
+      [
+        'equal version',
+        {
+          firstVersion: '1.1.1.1',
+          secondVersion: '1.1.1.1',
+          expectedValue: false,
+        },
+      ], [
+        'greater minor version',
+        {
+          firstVersion: '10.1.9.2',
+          secondVersion: '10.0.9.2',
+          expectedValue: false,
+        },
+      ], [
+        'greater patch version',
+        {
+          firstVersion: '1.1.1.2',
+          secondVersion: '1.1.1.1',
+          expectedValue: false,
+        },
+      ], [
+        'less than',
+        {
+          firstVersion: '2.0.0.0',
+          secondVersion: '2.0.0.1',
+          expectedValue: true,
+        },
+      ],
+    ])('%s', (_name, testcase) => {
+      const { firstVersion, secondVersion, expectedValue } = testcase;
+
+      // WHEN
+      const lhs = Version.parse(firstVersion);
+      const result = lhs.isLessThan(Version.parse(secondVersion));
+
+      expect(result).toEqual(expectedValue);
+    });
+  });
+
+  describe('.isLessThan constructor', () => {
+
+    // WHEN
+    const lhs = new Version([10, 0, 9, 2]);
+    const result = lhs.isLessThan(Version.parse('10.1.9.2'));
+
+    expect(result).toEqual(true);
+  });
+
+  describe('.parse', () => {
+    test.each<[string, { version: string, expectedValue: boolean }]>([
+      [
+        'ending with .',
+        {
+          version: '10.1.9.',
+          expectedValue: false,
+        },
+      ], [
+        'empty string',
+        {
+          version: '',
+          expectedValue: false,
+        },
+      ], [
+        'correct version',
+        {
+          version: '10.1.9.2',
+          expectedValue: true,
+        },
+      ],
+    ])('%s', (_name, testcase) => {
+      const { version, expectedValue } = testcase;
+      // WHEN
+      let result = false;
+      try {
+        Version.parse(version);
+        result = true;
+      } catch (error) {
+      } finally {
+      }
+
+      expect(result).toEqual(expectedValue);
```

Comment:
nit: while this test is technically valid, jest has some nice helpers for asserting exceptions being thrown that cleans up the `try`/`catch` block:

```ts
// Function to call
function parse() {
  Version.parse(version);
}

if (expectException) {
  expect(parse).toThrow();
}
else {
  expect(parse).not.toThrow();
}

```

Reply:
updated.